### PR TITLE
Renovate Shibuya and add rocstar demo

### PR DESCRIPTION
**Pull request by chrisli30**

### CODE REVIEW

File: src/rocstar.js [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+// It is defined as a UnitWeightCost variable in runtime.
+const TURING_INSTRUCTION_WEIGHT = 1000000000;
+const MIN_BALANCE_IN_PROXY = 10; // The proxy accounts are to be topped up if its balance fails below this number
+const SHIBUYA_TOKEN_ID_ON_TURING = 4;
+const TASK_FREQUENCY = 3600;
+const LISTEN_EVENT_DELAY = 3 * 60;
+
+const keyring = new Keyring({ type: 'sr25519' });
+
+const scheduleTask = async ({
```
Comment (chrisli30) [inline src/rocstar.js:22]: @imstar15 Rocstar needs to re-use the code of Shibuya. Please see #26 for my change suggestions.

File: src/rocstar.js [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+import { sendExtrinsic, getDecimalBN, listenEvents } from './common/utils';
+import { TuringStaging, Rocstar } from './config';
+import Account from './common/account';
+import 
+
+// One XCM operation is 1_000_000_000 weight - almost certainly a conservative estimate.
+// It is defined as a UnitWeightCost variable in runtime.
+const TURING_INSTRUCTION_WEIGHT = 1000000000;
+const MIN_BALANCE_IN_PROXY = 10; // The proxy accounts are to be topped up if its balance fails below this number
+const SHIBUYA_TOKEN_ID_ON_TURING = 4;
```
Comment (chrisli30) [inline src/rocstar.js:16]: I request to change this #27

File: src/rocstar.js [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+
+    const json = await readMnemonicFromFile();
+    const keyPair = keyring.addFromJson(json);
+    keyPair.unlock(process.env.PASS_PHRASE);
+
+    const account = new Account(keyPair);
+    await account.init([turingHelper, mangataHelper]);
+    account.print();
+
+    const parachainAddress = account.getChainByName(parachainName)?.address;
```
Comment (chrisli30) [inline src/rocstar.js:116]: Use `parachainAddress` instead of `sbyAddress` or `sdnAddress` so we can re-use the code.

File: src/rocstar.js [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+        const reserveTransferAssetsExtrinsic = shibuyaHelper.createReserveTransferAssetsExtrinsic(turingHelper.config.paraId, proxyAccountId, topUpAmountBN);
+        await sendExtrinsic(shibuyaHelper.api, reserveTransferAssetsExtrinsic, keyPair);
+    } else {
+        const freeBalanceOnTuring = (new BN(balanceOnTuring.free)).div(decimalBN);
+        console.log(`\nb) Proxy’s balance is ${freeBalanceOnTuring.toString()}, no need to top it up with reserve transfer ...`);
+    }
+
+    console.log(`\n3. Execute an XCM from ${parachainName} to schedule a task on ${turingChainName} ...`);
+
+    const result = await scheduleTask({
```
Comment (chrisli30) [inline src/rocstar.js:185]: I haven’t tested this rocstar.js code. @imstar15 could you test it? Load your wallet json to this program, and send me your wallet address for some Rocstar.

File: src/shibuya.js [hunk 5]
```diff
@@ -99,83 +98,101 @@ const main = async () => {
...
-        const freeSBYOnTuring = (new BN(sbyBalanceOnTuring.free)).div(sbyDecimalBN);
-        console.log(`\nb) Proxy’s balance is ${freeSBYOnTuring.toString()}, no need to top it up with SBY reserve transfer ...`);
+        const freeBalanceOnTuring = (new BN(balanceOnTuring.free)).div(decimalBN);
+        console.log(`\nb) Proxy’s balance is ${freeBalanceOnTuring.toString()}, no need to top it up with reserve transfer ...`);
     }
 
-    console.log('\n3. Execute an XCM from Shibuya to schedule a task on Turing ...');
+    console.log(`\n3. Execute an XCM from ${parachainName} to schedule a task on ${turingChainName} ...`);
+
     const result = await scheduleTask({
```
Comment (chrisli30) [inline src/shibuya.js:188]: shibuya.js is tested and working.

File: src/common/turingHelper.js [hunk 6]
```diff
@@ -93,6 +92,13 @@ class TuringHelper {
         const token = _.find(this.assets, { symbol });
         return token.decimals;
     }
+
+    getSiblingAssetId = async (paraId) => {
```
Comment (chrisli30) [inline src/common/turingHelper.js:96]: getAssetIdByLocation

**Review by imstar15 — APPROVED** [review]
Looks good to me!
Let me test later.

**Pull request merged**